src/test/java/teammates/logic/core/FeedbackResponseCommentsLogicTest.java

```diff
@@ -346,6 +353,288 @@ public void testGetFeedbackResponseCommentsForQuestionInSection_withSectionName_
+                new ArrayList<>(dataBundle.instructors.values()));
+        List<StudentAttributes> studentsInTeam =
+                roster.getTeamToMembersTable().getOrDefault(student.getTeam(), Collections.emptyList());
+        Set<String> studentsEmailInTeam = studentsInTeam.stream()
+                .map(StudentAttributes::getEmail)
+                .collect(Collectors.toSet());
+
+        FeedbackResponseAttributes response = dataBundle.feedbackResponses.get("response1ForQ1S1C1");
+        FeedbackQuestionAttributes relatedQuestion = dataBundle.feedbackQuestions.get("qn1InSession1InCourse1");
+        FeedbackResponseCommentAttributes relatedComment = FeedbackResponseCommentAttributes.builder()
+                .withCommentGiver("instructor1@course1.tmt")
+                .withCommentGiverType(FeedbackParticipantType.INSTRUCTORS)
+                .withVisibilityFollowingFeedbackQuestion(false)
+                .withCourseId("idOfTypicalCourse1")
+                .withFeedbackSessionName("First feedback session")
+                .withFeedbackResponseId("2%student2InCourse1@gmail.tmt%student5InCourse1@gmail.tmt")
+                .withShowCommentTo(Arrays.asList(FeedbackParticipantType.STUDENTS))
+                .build();
+
+        ______TS("failure: response is null");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, null, relatedQuestion, relatedComment));
+
+        ______TS("failure: related question is null");
+        assertFalse(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, null, relatedComment));
+
+        ______TS("success: giver is instructor; show comment to student; comment is visible to recipient");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student1InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is instructor; show comment to student; comment is visible to teammates");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student4InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("student3InCourse1@gmail.tmt", UserRole.STUDENT,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        ______TS("success: giver is instructor; comment is visible to giver by default");
+        assertTrue(frcLogic.isResponseCommentVisibleForUser("instructor1@course1.tmt", UserRole.INSTRUCTOR,
+                student, studentsEmailInTeam, response, relatedQuestion, relatedComment));
+
+        relatedComment = FeedbackResponseCommentAttributes.builder()
```

Comment:
we can improve this variable name to specify what it is about the comment that makes it special instead of `relatedComment`

Reply:
All comment variables names are updated to reflect their meaning.